Assets/MRTK/Providers/WindowsMixedReality/XR2018/WindowsMixedRealityDeviceManager.cs

```diff
@@ -688,13 +741,21 @@ private BaseWindowsMixedRealitySource GetOrAddController(InteractionSource inter
                 }
                 else if (interactionSource.kind == InteractionSourceKind.Controller)
                 {
-                    detectedController = new WindowsMixedRealityController(TrackingState.NotTracked, controllingHand, inputSource);
-                    if (!detectedController.Enabled)
+#if HP_CONTROLLER_ENABLED
+                    if (isHPController)
                     {
-                        // Controller failed to be setup correctly.
-                        // Return null so we don't raise the source detected.
-                        return null;
+                        // Add the controller as a HP Motion Controller
+                        detectedController = new HPMotionController(TrackingState.NotTracked, controllingHand, inputSource);
+                        if(trackedMotionControllerStates.ContainsKey(controllerId))
+                            ((HPMotionController)detectedController).MotionControllerState = trackedMotionControllerStates[controllerId];
```

Comment:
Also wonder if we can initially put the original object (line 748) in a local variable of the correct type, so we don't have to cast here, then can assign to `detectedController` at the end?

Reply:
Maybe not super important, since this is a one-time call when the controller is detected, but a thought